File: palpite.rb
numeros_sorteados = []

6.times do
    numero = rand(1..60)
    while numeros_sorteados.include?(numero)
        numero = rand(1..60)
    end
    numeros_sorteados << numero
end
numeros_ordenados = numeros_sorteados.sort
puts "Palpite para Mega-Sena\n"
puts ""
print "As dezenas sorteadas são: "
numeros_sorteados.each_with_index do |num, i|
    print "#{num}"
    if i < numeros_sorteados.length-1
        print " - "
    end
end
puts ""
print "As dezenas na ordem são: "
numeros_ordenados.each_with_index do |num, i|
    print "#{num}"
    if i < numeros_ordenados.length-1
        print " - "
    end
end
puts ""

File: palpite.rs
use rand::Rng;

fn gerar_dezenas() {
    let mut numeros = vec![];
    let mut rng = rand::thread_rng();
    for _ in 0..6 {
        let mut numero = rng.gen_range(1, 61);
        while numeros.contains(&numero) {
            numero = rng.gen_range(1, 61);
        }
        numeros.push(numero);
    }
    numeros.sort();
    println!("Palpite para Mega-Sena");
    print!("As dezenas sorteadas são: ");
    for i in 0..numeros.len() {
        print!("{}", numeros[i]);
        if i < numeros.len()-1 {
            print!(" - ");
        }
    }
    println!("\nAs dezenas na ordem são: ");
    for i in 0..numeros.len() {
        print!("{}", numeros[i]);
        if i < numeros.len()-1 {
            print!(" - ");
        }
    }
    println!("");
}

fn main() {
    gerar_dezenas();
}
